packages/rpc-core/src/rpc-subscriptions/__typetests__/slots-updates-notifications-type-test.ts

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+
+import { RpcSubscriptions } from '@solana/rpc-transport/dist/types/json-rpc-types';
+
+import { Slot } from '../../rpc-methods/common';
+import { UnixTimestamp } from '../../unix-timestamp';
+import { SolanaRpcSubscriptions, SolanaRpcSubscriptionsUnstable } from '../index';
+
+async () => {
+    const rpcSubscriptions = null as unknown as RpcSubscriptions<
+        SolanaRpcSubscriptions & SolanaRpcSubscriptionsUnstable
+    >;
+    const slotNotifications = await rpcSubscriptions.slotsUpdatesNotifications().subscribe();
+
+    slotNotifications satisfies AsyncIterable<
+        Readonly<{
+            parent?: Slot;
+            slot: Slot;
+            timestamp: UnixTimestamp;
+            type:
+                | 'completed'
+                | 'createdBank'
+                | 'dead'
+                | 'firstShredReceived'
+                | 'frozen'
+                | 'optimisticConfirmation'
+                | 'root';
+        }>
+    >;
```

Comment:
I think we'd be better off taking the approach that at every juncture the thing is what we think it is.
```suggestion
    type TNotification = Readonly<{
        parent?: Slot;
        slot: Slot;
        timestamp: UnixTimestamp;
        type:
            | 'completed'
            | 'createdBank'
            | 'dead'
            | 'firstShredReceived'
            | 'frozen'
            | 'optimisticConfirmation'
            | 'root';
    }>;
    rpcSubscriptions
        .slotsUpdatesNotifications() satisfies Promise<PendingRpcSubscription<TNotification>>;
    rpcSubscriptions    
        .slotsUpdatesNotifications()
        .subscribe() satisfies Promise<AsyncIterable<TNotification>>;
```